### PR TITLE
Add text inputs of different widths

### PR DESCRIPTION
**Pull request by sukhrajghuman**
adding option for different sized text inputs. This will also mean our text inputs will be the same width on all browsers, since previously we were using the browser default widths.

image

### CODE REVIEW

File: packages/text-inputs/src/sass/_module.scss [hunk 1]
```diff
@@ -108,6 +108,23 @@
...
+	&.au-text-input--width-2 {
+		@include AU-space( max-width, 4em );
+	}
+	
+	//4 character text box. i.e postcode
+	&.au-text-input--width-4 {
+		@include AU-space( max-width, 5em );
+	}
+
+	//4 character text box. i.e postcode
```
Comment (adamzerella) [inline packages/text-inputs/src/sass/_module.scss]:
```suggestion
	//10 character text box. i.e phone number
```

File: packages/text-inputs/src/sass/_module.scss [hunk 2]
```diff
@@ -174,4 +191,3 @@ textarea.au-text-input {
 	max-width: 100%;
 	min-height: 6em;
 }
```
Comment (adamzerella) [inline packages/text-inputs/src/sass/_module.scss]:
```suggestion
}

```

File: packages/text-inputs/tests/react/index.js [hunk 3]
```diff
@@ -71,6 +71,15 @@ ABCZ abcjg liI1 12340" />
 			<h2>Textarea default value disabled</h2>
 			<label htmlFor="8">Hello</label>
 			<AUtextInput id="8" defaultValue="abc" as="textarea" disabled />
+
+			<hr />
+			<h2>Text inputs different widths</h2>
```
Comment (adamzerella) [inline packages/text-inputs/tests/react/index.js]: Add a quick test for using the { number } type

File: packages/text-inputs/tests/site/index.html [hunk 4]
```diff
@@ -185,6 +185,17 @@ <h2>Textarea input, using <code>body</code></h2>
...
+
+			<hr>
+			<h3>Text inputs different widths</h3>
+			<label class="label" for="width-2">Width 2</label>
+			<input type="text" value="07"  class="au-text-input au-text-input--width-2" id="width-2"/>
+			<label class="label" for="width-4">Width 4</label>
+			<input type="text" value="2600" class="au-text-input au-text-input--width-4" id="width-4"/>
+			<label class="label" for="width-10">Width 10</label>
+			<input type="text" value="0400 000 000" class="au-text-input au-text-input--width-10" id="width-10"/>
+			<label class="label" for="normal">Normal width</label>
```
Comment (adamzerella) [inline packages/text-inputs/tests/site/index.html]:
```suggestion
			<label class="label" for="default">Default width</label>
```

File: packages/text-inputs/tests/site/index.html [hunk 5]
```diff
@@ -185,6 +185,17 @@ <h2>Textarea input, using <code>body</code></h2>
...
+			<input type="text" value="+61 400 000 000" class="au-text-input" id="normal"/>
```
Comment (adamzerella) [inline packages/text-inputs/tests/site/index.html]:
```suggestion
			<input type="text" value="+61 400 000 000" class="au-text-input" id="default"/>
```

File: packages/text-inputs/tests/site/index.html [hunk 6]
```diff
@@ -321,6 +332,17 @@ <h2>Textarea input, using <code>body</code></h2>
...
+
+			<hr>
+			<h3>Text inputs different widths</h3>
+			<label class="label" for="width-2-dark">Width 2</label>
+			<input type="text" value="07" class="au-text-input au-text-input--dark au-text-input--width-2" id="width-2-dark"/>
+			<label class="label" for="width-4-dark">Width 4</label>
+			<input type="text" value="2600" class="au-text-input au-text-input--dark au-text-input--width-4" id="width-4-dark"/>
+			<label class="label" for="width-10-dark">Width 10</label>
+			<input type="text" value="0400 000 000" class="au-text-input au-text-input--dark au-text-input--width-10" id="width-10-dark"/>
+			<label class="label" for="normal-width-dark">Normal</label>
```
Comment (adamzerella) [inline packages/text-inputs/tests/site/index.html]:
```suggestion
			<label class="label" for="default-width-dark">Default</label>
```

File: packages/text-inputs/tests/site/index.html [hunk 7]
```diff
@@ -321,6 +332,17 @@ <h2>Textarea input, using <code>body</code></h2>
...
+			<input type="text" value="+61 400 000 000" class="au-text-input au-text-input--dark" id="normal-width-dark"/>
```
Comment (adamzerella) [inline packages/text-inputs/tests/site/index.html]:
```suggestion
			<input type="text" value="+61 400 000 000" class="au-text-input au-text-input--dark" id="default-width-dark"/>
```

File: packages/text-inputs/tests/site/index.html [hunk 8]
```diff
@@ -460,6 +482,16 @@ <h2>Textarea input, using <code>body</code></h2>
...
 			</div>
+			<hr>
+			<h3>Text inputs different widths</h3>
+			<label class="label" for="width-2-alt">Width 2</label>
+			<input type="text" value="07" class="au-text-input au-text-input--alt au-text-input--width-2" id="width-2-alt"/>
+			<label class="label" for="width-4-alt">Width 4</label>
+			<input type="text" value="2600" class="au-text-input au-text-input--alt au-text-input--width-4" id="width-4-alt"/>
+			<label class="label" for="width-10-alt">Width 10</label>
+			<input type="text" value="0400 000 000" class="au-text-input au-text-input--alt au-text-input--width-10" id="width-10-alt"/>
+			<label class="label" for="normal-width-alt">Normal</label>
```
Comment (adamzerella) [inline packages/text-inputs/tests/site/index.html]:
```suggestion
			<label class="label" for="default-width-alt">Default</label>
```

File: packages/text-inputs/tests/site/index.html [hunk 9]
```diff
@@ -460,6 +482,16 @@ <h2>Textarea input, using <code>body</code></h2>
...
+			<input type="text" value="+61 400 000 000" class="au-text-input au-text-input--alt" id="normal-width-alt"/>
```
Comment (adamzerella) [inline packages/text-inputs/tests/site/index.html]:
```suggestion
			<input type="text" value="+61 400 000 000" class="au-text-input au-text-input--alt" id="default-width-alt"/>
```

File: packages/text-inputs/tests/site/index.html [hunk 10]
```diff
@@ -597,6 +629,16 @@ <h2>Textarea input, using <code>body</code></h2>
...
 			</div>
+			<hr>
+			<h3>Text inputs different widths</h3>
+			<label class="label" for="width-2-dark-alt">Width 2</label>
+			<input type="text" value="07" class="au-text-input au-text-input--dark au-text-input--width-2" id="width-2-dark-alt"/>
+			<label class="label" for="width-4-dark-alt">Width 4</label>
+			<input type="text" value="2600" class="au-text-input au-text-input--dark au-text-input--width-4" id="width-4-dark-alt"/>
+			<label class="label" for="width-10-dark-alt">Width 10</label>
+			<input type="text" value="0400 000 000" class="au-text-input au-text-input--dark au-text-input--width-10" id="width-10-dark-alt"/>
+			<label class="label" for="normal-width-dark-alt">Normal</label>
```
Comment (adamzerella) [inline packages/text-inputs/tests/site/index.html]:
```suggestion
			<label class="label" for="default-width-dark-alt">Default</label>
```

File: packages/text-inputs/tests/site/index.html [hunk 11]
```diff
@@ -597,6 +629,16 @@ <h2>Textarea input, using <code>body</code></h2>
...
+			<input type="text" value="+61 400 000 000" class="au-text-input au-text-input--dark" id="normal-width-dark-alt"/>
```
Comment (adamzerella) [inline packages/text-inputs/tests/site/index.html]:
```suggestion
			<input type="text" value="+61 400 000 000" class="au-text-input au-text-input--dark" id="default-width-dark-alt"/>
```

**Comment by adamzerella** [comment]
I think add another input for `width-3` and we're golden 👍

File: packages/text-inputs/src/sass/_module.scss [hunk 12]
```diff
@@ -42,7 +42,8 @@
 	border-radius: $AU-border-radius;
 	vertical-align: middle;
 	appearance: none;
-
+	width: 100%;
```
Comment (adamzerella) [inline packages/text-inputs/src/sass/_module.scss]:
```suggestion
```

File: packages/text-inputs/src/js/react.js [hunk 13]
```diff
@@ -58,6 +60,7 @@ AUtextInput.propTypes = {
 	block: PropTypes.bool,
 	status: PropTypes.oneOf([ 'valid', 'invalid' ]),
 	className: PropTypes.string,
+	width: PropTypes.oneOf(["xs", "sm", "md","lg", "xl"])
```
Comment (adamzerella) [inline packages/text-inputs/src/js/react.js]:
```suggestion
	width: PropTypes.oneOf([ 'xs', 'sm', 'md', 'lg', 'xl' ])
```

Try and keep the quote consistent

File: packages/text-inputs/src/sass/_module.scss (hunk 12 above)
Comment (sukhrajghuman) [reply packages/text-inputs/src/sass/_module.scss]: No we need this, if I remove it here then I need to add it in all of the different width classes.

**Pull request merged**